PuppeteerSharp.Dom/DomHandle.cs

```diff
@@ -218,6 +209,104 @@ Task<T> IJSHandle.JsonValueAsync<T>()
             return Handle.JsonValueAsync<T>();
         }
 
+        /// <summary>
+        /// Parse value of element to designated type
+        /// </summary>
+        /// <typeparam name="T">Designated type</typeparam>
+        /// <param name="value">Value element</param>
+        /// <returns>Value of element of designated type</returns>
+        /// <exception cref="InvalidOperationException">Value not found (null).</exception>
+        /// <exception cref="FormatException">Type not recognized and value was unable to be parsed.</exception>
+        public static T ParseJSValueTo<T>(JsonElement? value)
+        {
+            var returnType = typeof(T);
+            if (returnType == typeof(object) || returnType == value.GetType())
+            {
+                return (T)(object)value;
+            }
+            var valueIsNull = value == null || value.Value.ValueKind == JsonValueKind.Null || value.Value.ValueKind == JsonValueKind.Undefined;
+            var nullable = returnType.IsClass;
+            if (Nullable.GetUnderlyingType(returnType) != null)
+            {
+                nullable = true;
+                returnType = Nullable.GetUnderlyingType(returnType);
+            }
+
+            if (valueIsNull)
+            {
+                if (nullable)
+                {
+                    return (T)(object)null;
+                }
+                throw new InvalidOperationException($"Result of evaluating a script is null when type {typeof(T)} expected.");
+            }
+            else if (returnType == typeof(string))
+            {
+                return (T)(object)value.Value.GetString();
+            }
+            else if (returnType.IsEnum)
+            {
+                var typeConverter = TypeDescriptor.GetConverter(returnType);
+                return (T)typeConverter.ConvertFrom(value.ToString());
+            }
+            else if (returnType == typeof(int))
```

Comment:
Is there a reason to prefer handling the types individually with manual coercion rather than using `TypeDescriptor.GetConverter`?

From memory `TypeDescriptor` is extensible allowing you to use your own custom converter if required. It will also cover a much larger number of types.